prow/sidecar/censor_test.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sidecar
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"k8s.io/test-infra/prow/gcsupload"
+	"k8s.io/test-infra/prow/pod-utils/wrapper"
+	"k8s.io/test-infra/prow/secretutil"
+	"k8s.io/test-infra/prow/testutil"
+)
+
+func TestCensor(t *testing.T) {
+	preamble := func() string {
+		return `In my younger and more vulnerable years my father gave me some advice that I’ve been turning over in my mind ever since.`
+	}
+
+	var testCases = []struct {
+		name          string
+		input, output string
+		secrets       []string
+		frameSize     int
+	}{
+		{
+			name:      "input smaller than frame size",
+			input:     preamble()[:100],
+			secrets:   []string{"younger", "my"},
+			output:    "In ** ******* and more vulnerable years ** father gave me some advice that I’ve been turning over ",
+			frameSize: 200,
+		},
+		{
+			name:      "input larger than frame size, not a multiple",
+			input:     preamble()[:100],
+			secrets:   []string{"younger", "my"},
+			output:    "In ** ******* and more vulnerable years ** father gave me some advice that I’ve been turning over ",
+			frameSize: 16,
+		},
+	}
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			censorer := secretutil.NewCensorer()
+			censorer.Refresh(testCase.secrets...)
+			input := ioutil.NopCloser(bytes.NewBufferString(testCase.input))
```

Comment:
Can you also verify that `Close` is being called?

Reply:
Not trivially, no, I don't think I can. Why?